lib/plugin.js

```diff
@@ -184,6 +185,14 @@ const setupProgress = (axios) => {
   axios.defaults.onDownloadProgress = onProgress
 }<% } %>
 
+if (process.server) {
```

Comment:
Why this change is necessary with this feature?

Reply:
This is to extend the `process.axios` instance with extra methods and configuration specific to this module.

As I've hinted in the commit message, we could also extract the "extending code" into a separate file and use it both from the module and the plugin but that would require refactoring to remove the use of templates. And that means we couldn't include parts of its code conditionally, based on settings. The whole code would always be included in the bundles.

Reply:
I think we can write node specific logic to create instance out of the plugin so we have clear control over behavior and also not depending on first SSR request to be done to make instance initialized

Reply:
What do you mean by creating an instance out of the plugin?

Extract the code into a separate file as I've suggested or just write extra code? We basically need this part accessible from the module also: https://github.com/nuxt-community/axios-module/blob/a3f708d4b2e2afa08dcedbfe58b7600c23635fde/lib/plugin.js#L6-L113
So are you suggesting to duplicate it or extract to a separate file or something else?

Reply:
We may refactor useful logic into a shared CJS file as well and use both for module and runtime to avoid duplicate code like [this](https://github.com/nuxt-community/pwa-module/blob/master/lib/meta/meta.merge.js)

Reply:
Yes. The only concern with that is that means the whole code will be included in the bundles rather than only code that is needed for enabled options. But I guess you are fine with that?

Reply:
ES6 / Module syntax is possible for nuxt modules with esm which is deprecated. We also disable esm support for typescript and test environments. When testing another project that depends on axios module, this can be problematic.

> Yes. The only concern with that is that means the whole code will be included in the bundles rather than only code that is needed for enabled options. But I guess you are fine with that?

We can mark cjs file as esm and use pure comment to allow tree-shaking per export but it shouldn't be a big deal for small shared utils.

I think ideally we should start migrating modules to typescript and proper build step to avoid such issues and also share logic in an easier way..